indexing-service/src/main/java/org/apache/druid/indexing/common/TaskReport.java

```diff
@@ -48,13 +51,29 @@ public interface TaskReport
   /**
    * Returns an order-preserving map that is suitable for passing into {@link TaskReportFileWriter#write}.
    */
-  static Map<String, TaskReport> buildTaskReports(TaskReport... taskReports)
+  static ReportMap buildTaskReports(TaskReport... taskReports)
   {
-    // Use LinkedHashMap to preserve order of the reports.
-    Map<String, TaskReport> taskReportMap = new LinkedHashMap<>();
+    ReportMap taskReportMap = new ReportMap();
     for (TaskReport taskReport : taskReports) {
       taskReportMap.put(taskReport.getReportKey(), taskReport);
     }
     return taskReportMap;
   }
+
+  /**
+   * Represents an ordered map from report key to a TaskReport that is compatible
+   * for writing out reports to files or serving over HTTP.
+   * <p>
+   * This class is needed for Jackson serde to work correctly. Without this class,
+   * a TaskReport is serialized without the type information and cannot be
+   * deserialized back into a concrete implementation.
+   */
+  class ReportMap extends LinkedHashMap<String, TaskReport>
```

Comment:
Are there any tests that verify the reports are indeed ordered since we rely on a `LinkedHashMap`? Just looking at the callers of `buildTaskReports()`, I don't seem to find any.

Reply:
No, I can add a test to verify the order. Although, I don't see any actual task writing a report map that contains multiple entries.